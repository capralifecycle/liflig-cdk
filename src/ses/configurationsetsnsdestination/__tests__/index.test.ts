import { App, Stack } from "aws-cdk-lib"
import * as sns from "aws-cdk-lib/aws-sns"
import "jest-cdk-snapshot"
import { ConfigurationSetSnsDestination } from "../index"

test("configuration-set-sns-destination", () => {
  const app = new App()
  const stack = new Stack(app, "Stack")

  const snsTopic = new sns.Topic(stack, "Topic")

  new ConfigurationSetSnsDestination(stack, "SnsDestination", {
    configurationSetName: "exampleconfigurationset",
    eventDestinationName: "sns",
    snsTopic: snsTopic,
    matchingEventTypes: [
      "BOUNCE",
      "COMPLAINT",
      "DELIVERY",
      "DELIVERY_DELAY",
      "REJECT",
      "SEND",
    ],
  })

  expect(stack).toMatchCdkSnapshot({
    ignoreAssets: true,
  })
})
