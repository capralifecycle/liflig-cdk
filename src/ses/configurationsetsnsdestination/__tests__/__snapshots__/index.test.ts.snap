// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`configuration-set-sns-destination 1`] = `
Object {
  "Resources": Object {
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": Object {
      "DependsOn": Array [
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
      ],
      "Properties": Object {
        "Code": Any<Object>,
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
            "Arn",
          ],
        },
        "Runtime": "nodejs22.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:PutRetentionPolicy",
                "logs:DeleteRetentionPolicy",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "Roles": Array [
          Object {
            "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SnsDestinationEventsHandler7DFCB014": Object {
      "DependsOn": Array [
        "SnsDestinationEventsHandlerServiceRole94495B0D",
      ],
      "Properties": Object {
        "Code": Any<Object>,
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "SnsDestinationEventsHandlerServiceRole94495B0D",
            "Arn",
          ],
        },
        "Runtime": "nodejs22.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "SnsDestinationEventsHandlerAllowInvokeStackTopic2F4048F0389B8B0A": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "SnsDestinationEventsHandler7DFCB014",
            "Arn",
          ],
        },
        "Principal": "sns.amazonaws.com",
        "SourceArn": Object {
          "Ref": "TopicBFC7AF6E",
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "SnsDestinationEventsHandlerLogRetention7511E399": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "SnsDestinationEventsHandler7DFCB014",
              },
            ],
          ],
        },
        "RetentionInDays": 90,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "SnsDestinationEventsHandlerServiceRole94495B0D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "SnsDestinationEventsHandlerTopic02278A9D": Object {
      "Properties": Object {
        "Endpoint": Object {
          "Fn::GetAtt": Array [
            "SnsDestinationEventsHandler7DFCB014",
            "Arn",
          ],
        },
        "Protocol": "lambda",
        "TopicArn": Object {
          "Ref": "TopicBFC7AF6E",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "SnsDestinationF568367B": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "ConfigurationSetName": "exampleconfigurationset",
        "EventDestinationName": "sns",
        "MatchingEventTypes": Array [
          "BOUNCE",
          "COMPLAINT",
          "DELIVERY",
          "DELIVERY_DELAY",
          "REJECT",
          "SEND",
        ],
        "Serial": 1,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "lifligcdkconfigurationsetsnsdestinationProviderframeworkonEvent102A5A7F",
            "Arn",
          ],
        },
        "SnsTopicArn": Object {
          "Ref": "TopicBFC7AF6E",
        },
      },
      "Type": "AWS::CloudFormation::CustomResource",
      "UpdateReplacePolicy": "Delete",
    },
    "TopicBFC7AF6E": Object {
      "Type": "AWS::SNS::Topic",
    },
    "lifligcdkconfigurationsetsnsdestinationFunction24D2C768": Object {
      "DependsOn": Array [
        "lifligcdkconfigurationsetsnsdestinationFunctionServiceRoleDefaultPolicy848780D4",
        "lifligcdkconfigurationsetsnsdestinationFunctionServiceRoleDAE79368",
      ],
      "Properties": Object {
        "Code": Any<Object>,
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "lifligcdkconfigurationsetsnsdestinationFunctionServiceRoleDAE79368",
            "Arn",
          ],
        },
        "Runtime": "nodejs22.x",
        "Timeout": 300,
      },
      "Type": "AWS::Lambda::Function",
    },
    "lifligcdkconfigurationsetsnsdestinationFunctionServiceRoleDAE79368": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "lifligcdkconfigurationsetsnsdestinationFunctionServiceRoleDefaultPolicy848780D4": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ses:CreateConfigurationSetEventDestination",
                "ses:UpdateConfigurationSetEventDestination",
                "ses:DeleteConfigurationSetEventDestination",
                "ses:GetConfigurationSetEventDestinations",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "lifligcdkconfigurationsetsnsdestinationFunctionServiceRoleDefaultPolicy848780D4",
        "Roles": Array [
          Object {
            "Ref": "lifligcdkconfigurationsetsnsdestinationFunctionServiceRoleDAE79368",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "lifligcdkconfigurationsetsnsdestinationProviderframeworkonEvent102A5A7F": Object {
      "DependsOn": Array [
        "lifligcdkconfigurationsetsnsdestinationProviderframeworkonEventServiceRoleDefaultPolicyF016E55B",
        "lifligcdkconfigurationsetsnsdestinationProviderframeworkonEventServiceRoleC2624999",
      ],
      "Properties": Object {
        "Code": Any<Object>,
        "Description": "AWS CDK resource provider framework - onEvent (Stack/liflig-cdk.configuration-set-sns-destination/Provider)",
        "Environment": Object {
          "Variables": Object {
            "USER_ON_EVENT_FUNCTION_ARN": Object {
              "Fn::GetAtt": Array [
                "lifligcdkconfigurationsetsnsdestinationFunction24D2C768",
                "Arn",
              ],
            },
          },
        },
        "Handler": "framework.onEvent",
        "LoggingConfig": Object {
          "ApplicationLogLevel": "FATAL",
          "LogFormat": "JSON",
        },
        "Role": Object {
          "Fn::GetAtt": Array [
            "lifligcdkconfigurationsetsnsdestinationProviderframeworkonEventServiceRoleC2624999",
            "Arn",
          ],
        },
        "Runtime": "nodejs22.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "lifligcdkconfigurationsetsnsdestinationProviderframeworkonEventServiceRoleC2624999": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "lifligcdkconfigurationsetsnsdestinationProviderframeworkonEventServiceRoleDefaultPolicyF016E55B": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "lifligcdkconfigurationsetsnsdestinationFunction24D2C768",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "lifligcdkconfigurationsetsnsdestinationFunction24D2C768",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "lambda:GetFunction",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "lifligcdkconfigurationsetsnsdestinationFunction24D2C768",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "lifligcdkconfigurationsetsnsdestinationProviderframeworkonEventServiceRoleDefaultPolicyF016E55B",
        "Roles": Array [
          Object {
            "Ref": "lifligcdkconfigurationsetsnsdestinationProviderframeworkonEventServiceRoleC2624999",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
